src/OrchardCore.Modules/OrchardCore.Media/Services/AttachedMediaFieldsFolderAuthorizationHandler.cs

```diff
@@ -11,21 +14,24 @@ namespace OrchardCore.Media.Services
     /// Check if the path passed as resource is inside the AttachedMediaFieldsFolder
     /// and in case it is, It checks if the user has ManageAttachedMediaFieldsFolder permission
     /// </summary>
-    public class AttachedMediaFieldsFolderAuthorizationHandler : AuthorizationHandler<PermissionRequirement>
+    public class AttachedMediaFieldsFolderAuthorizationHandler : AuthorizationHandler<PermissionRequirement> 
```

Comment:
I am sorry I don't understand the purpose of original AttachedMediaFieldsFolderAuthorizationHandler. it seems to check the resource if under the path "media/mediafields". but under what circumstances will folder "media/mediafields" be created ? Can you explain more this ? Thank you.

Reply:
It's solely related to the Attached Media Field editor option on the media field (go to settings for a Media Field and select Attached as the editor).

When you use it, files are uploaded and selected, on an individual basis for the content item, rather than having the entire media library available.

So this permission handler is to allow management of the entire folder structure that gets created under it.

Does that make sense?

Reply:
Thank you. I have refactored AttachedMediaFieldsFolderAuthorizationHandler and rename to ManageMediaFolderAuthorizationHandler, let me know if any improvement need. thank you.